client/components/translate/translate.config.js

```diff
@@ -73,65 +121,32 @@ angular.module("managerApp")
      *        ]
      *    });
      */
-    this.add = function ($stateProvider) {
-
-        $stateProvider.decorator("translations", function (state) {
-            var routeOption = state.self;
-
-            if (routeOption.translations) {
-
-                var templateUrlTab = [];
-                var translationsTab = routeOption.translations;
-
-                if (routeOption.templateUrl) {
-                    templateUrlTab.push(routeOption.templateUrl);
-                }
-
-                if (routeOption.views) {
-                    angular.forEach(routeOption.views, function (value) {
-                        if (value.templateUrl) {
-                            templateUrlTab.push(value.templateUrl);
-                        }
-                        if (value.translations) {
-                            translationsTab = _.union(translationsTab, value.translations);
+        this.add = function ($transitionsProvider, $stateProvider) {
+
+            $stateProvider.decorator("translations", state => {
+                const translations = _.chain(getAncestors(state))
```

Comment:
@frenautvh I've made something quite different from https://github.com/ovh-ux/ovh-manager-web/pull/335 because I wanted to have the exact same behavior as before. For exemple, if I'm in iaas.pci-project.compute.infratructure.diagram I want to have all the translations of iaas.pci-project.compute.infrastructure and not only what is in /cloud/project/compute/infrastructure/translations.
I've also kept the $stateProvider so that the list of required translations is loaded during the first application load. Don't worry, the JSON themselves are still lazy-loaded. :)